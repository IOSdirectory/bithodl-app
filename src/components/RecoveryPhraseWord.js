import React, { Component } from 'react';
import { StyleSheet, View } from 'react-native';
import PropTypes from 'prop-types';
import StyledText from './StyledText';

const styles = StyleSheet.create({
  container: {
    width: '49%',
    padding: 4
  },
  wrapper: {
    backgroundColor: '#fbfbfb',
    borderColor: '#DDDDDF',
    borderWidth: 1,
    borderRadius: 30,
    padding: 7,
    paddingLeft: 12
  },
  word: {
    fontFamily: 'Menlo-Regular',
    fontSize: 11
  }
});

export default class RecoveryPhraseWord extends Component {
  render() {
    const index = this.props.index;
    const word = this.props.word;

    return (
      <View style={styles.container}>
        <View style={styles.wrapper}>
          <StyledText style={styles.word}>
            {index + 1}. {word}
          </StyledText>
        </View>
      </View>
    );
  }
}

RecoveryPhraseWord.propTypes = {
  index: PropTypes.number.isRequired,
  word: PropTypes.string.isRequired
};
