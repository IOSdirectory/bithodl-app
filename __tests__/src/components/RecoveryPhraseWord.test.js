import React from 'react';
import renderer from 'react-test-renderer';
import RecoveryPhraseWord from '../../../src/components/RecoveryPhraseWord';

describe('RecoveryPhraseWord', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <RecoveryPhraseWord index={10} word='bulb' />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
