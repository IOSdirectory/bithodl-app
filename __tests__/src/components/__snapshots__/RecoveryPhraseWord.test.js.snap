// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoveryPhraseWord renders correctly 1`] = `
<View
  style={
    Object {
      "padding": 4,
      "width": "49%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fbfbfb",
        "borderColor": "#DDDDDF",
        "borderRadius": 30,
        "borderWidth": 1,
        "padding": 7,
        "paddingLeft": 12,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "fontFamily": "Menlo-Regular",
            "fontSize": 11,
          },
        ]
      }
    >
      11
      . 
      bulb
    </Text>
  </View>
</View>
`;
